### PR TITLE
Enable service typesupport for C

### PR DESCRIPTION
**Pull request by jacquelinekay**
Connects to ros2/rcl#27

depends on #120

### CODE REVIEW

**Comment by jacquelinekay** [comment]
OSX seems to have problems with the generated C code included the C++ code. Looking into it.

http://ci.ros2.org/job/ci_osx/872/console#console-section-0

File: rosidl_typesupport_opensplice_c/cmake/rosidl_typesupport_opensplice_c_generate_interfaces.cmake [hunk 1]
```diff
@@ -267,6 +269,7 @@ ament_target_dependencies(
   ${rosidl_generate_interfaces_TARGET}${_target_suffix}
   "OpenSplice"
   "rosidl_typesupport_opensplice_c")
+  #"${PROJECT_NAME}__rosidl_generator_c")
```
Comment (dirk-thomas) [inline rosidl_typesupport_opensplice_c/cmake/rosidl_typesupport_opensplice_c_generate_interfaces.cmake]: What about this comment?

File: rosidl_typesupport_opensplice_c/resource/srv__type_support_c.cpp.template [hunk 2]
```diff
@@ -0,0 +1,168 @@
+// generated from rosidl_typesupport_opensplice_c/resource/srv__type_support.cpp.template
```
Comment (dirk-thomas) [inline rosidl_typesupport_opensplice_c/resource/srv__type_support_c.cpp.template]: Needs "no copyright" line.

Comment (dirk-thomas) [inline rosidl_typesupport_opensplice_c/resource/srv__type_support_c.cpp.template]: Never mind for cpp files.

File: rosidl_typesupport_opensplice_c/cmake/rosidl_typesupport_opensplice_c_generate_interfaces.cmake (hunk 1 above)
Comment (jacquelinekay) [inline rosidl_typesupport_opensplice_c/cmake/rosidl_typesupport_opensplice_c_generate_interfaces.cmake]: Removed

File: rosidl_typesupport_opensplice_c/cmake/rosidl_typesupport_opensplice_c_generate_interfaces.cmake [hunk 3]
```diff
@@ -179,6 +179,7 @@ configure_file(
   @ONLY
 )
 list(APPEND _generated_msg_files "${_visibility_control_file}")
+list(APPEND _generated_srv_files "${_visibility_control_file}")
```
Comment (dirk-thomas) [inline rosidl_typesupport_opensplice_c/cmake/rosidl_typesupport_opensplice_c_generate_interfaces.cmake]: Why does the file need to be in the `_generated_srv_files` list too?

File: rosidl_typesupport_opensplice_c/resource/srv__type_support_c.cpp.template [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
```
Comment (wjwwood) [inline rosidl_typesupport_opensplice_c/resource/srv__type_support_c.cpp.template]: I think this file name needs to be updated.

File: rosidl_typesupport_opensplice_c/resource/srv__type_support_c.cpp.template [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+// generated code does not contain a copyright notice
+
+@#########################################e#############################
+@# EmPy template for generating <srv>__type_support.cpp files
```
Comment (wjwwood) [inline rosidl_typesupport_opensplice_c/resource/srv__type_support_c.cpp.template]: Same.

**Comment by wjwwood** [comment]
Other than the comments it lgtm. Please remember to quash though (lots of small commits here).

**Pull request merged**